tools/linter/adapters/circleci_linter.py

```diff
@@ -0,0 +1,153 @@
+"""
+Checks that the configuration in .circleci/config.yml has been properly regenerated.
+"""
+
+import os
+import argparse
+import subprocess
+import sys
+import logging
+import time
+from enum import Enum
+from typing import List, NamedTuple, Optional
+import json
+
+
+CHECKED_IN_FILE = "config.yml"
+REGENERATION_SCRIPT = "regenerate.sh"
+
+PARENT_DIR = os.path.basename(os.path.dirname(os.path.abspath(__file__)))
+README_PATH = os.path.join(PARENT_DIR, "README.md")
+
+
+class LintSeverity(str, Enum):
```

Comment:
Is there some reason we can't share these between all the lint adapters? One of my main things with the python lint runners right now is they've gotten pretty complicated over time (i.e. we have 1000+ lines to run clang-tidy) so if we do something new it should be understandable / extensible